### PR TITLE
reviewed and test scroll messaging system

### PR DESCRIPTION
**Pull request by Jemiiah**
Resolves : #15 
1.  Added payable to functions using sendMessage in the ScrollStateBridge and removed hardcoded value of 0 to use msg.value
2.  Added refundAddress in functions that sendMessage incase of an error or potentially malicious behavior to prevent locked balances or failures
3.  xAdded Natspec where necessary

### CODE REVIEW

**Comment by Jemiiah** [comment]
@dragan2234 please kindly review

**Comment by dragan2234** [comment]
@Jemiiah Can you please fix the tests in `src/test/ScrollStateBridge.t.sol` since they are failing due to these changes?

**Comment by Jemiiah** [comment]
okay ser @dragan2234

**Comment by Jemiiah** [comment]
@dragan2234 done ser

**Comment by Jemiiah** [comment]
@dragan pls kindly review

**Comment by dragan2234** [comment]
@Jemiiah These tests are still failing for me when I run `make test`

```
Failing tests:
Encountered 3 failing tests in src/test/ScrollStateBridge.t.sol:ScrollStateBridgeTest
[FAIL. Reason: log != expected log; counterexample: calldata=0x04f2ce7200000000000000000000000006b1019ade235d527ed8f6aa75b596c005c80bab args=[38201316336125633280580617972413903052546247595 [3.82e46]]] test_owner_setRootHistoryExpiry_succeeds(uint256) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: log != expected log; counterexample: calldata=0xf944a4540000000000000000000000001f7ca6925ffb92da0144d0dea649174936fadfd1 args=[0x1F7cA6925fFB92Da0144D0dEa649174936fAdfd1]] test_owner_transferOwnershipOp_succeeds(address) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: log != expected log] test_propagateRoot_suceeds() (gas: 36155)
```

**Comment by Jemiiah** [comment]
@dragan2234 pls check your TG

File: src/test/ScrollStateBridge.t.sol [hunk 1]
```diff
@@ -184,7 +177,7 @@ contract ScrollStateBridgeTest is PRBTest, StdCheats {
         emit SetRootHistoryExpiry(_rootHistoryExpiry);
 
         vm.prank(owner);
-        scStateBridge.setRootHistoryExpiry(_rootHistoryExpiry);
+        scStateBridge.setRootHistoryExpiry{value: 2 ether}(_rootHistoryExpiry,0, owner);
```
Comment (dragan2234) [inline src/test/ScrollStateBridge.t.sol:180]: Why are we passing such a big number `2 ether` here? @Jemiiah

File: src/test/ScrollStateBridge.t.sol [hunk 2]
```diff
@@ -116,6 +108,7 @@ contract ScrollStateBridgeTest is PRBTest, StdCheats {
             new ScrollStateBridge(mockWorldIDAddress, opWorldIDAddress, opCrossDomainMessengerAddress);
 
         owner = scStateBridge.owner();
+        deal(owner, 100 ether);
```
Comment (dragan2234) [inline src/test/ScrollStateBridge.t.sol:111]: Why are we passing such a big number `100 ether` here? @Jemiiah

File: src/test/ScrollStateBridge.t.sol [hunk 3]
```diff
@@ -92,18 +92,10 @@ contract ScrollStateBridgeTest is PRBTest, StdCheats {
...
-        /// @notice Roll the fork to a block where both Optimim's and Base's crossDomainMessenger contract is deployed
-        /// @notice and the Base crossDomainMessenger ResolvedDelegateProxy target address is initialized
-        vm.rollFork(17711915);
-
-        if (block.chainid == 1) {
-            opCrossDomainMessengerAddress = address(0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1);
-        } else {
-            revert invalidCrossDomainMessengerFork();
-        }
+        opCrossDomainMessengerAddress = address(0x6774Bcbd5ceCeF1336b5300fb5186a12DDD8b367);
```
Comment (dragan2234) [inline src/test/ScrollStateBridge.t.sol]: @Jemiiah Shouldn't this be Scroll and not Op (optimism)?

Comment (Jemiiah) [reply src/test/ScrollStateBridge.t.sol:97]: yes @dragan2234 it's scroll address I forgot to rename it

**Comment by Jemiiah** [comment]
@dragan2234 made all necessary correction

**Review by dragan2234 — APPROVED** [review]
Thank you for the work

**Pull request merged**